src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialProvider.java

```diff
@@ -195,18 +195,32 @@ public void stopWatchingForSecrets() {
 
     @Override
     public <C extends Credentials> List<C> getCredentials(Class<C> type, ItemGroup itemGroup, Authentication authentication) {
-        LOG.log(Level.FINEST, "getCredentials called with type {0} and authentication {1}", new Object[] {type.getName(), authentication});
+        LOG.log(Level.FINEST, "getCredentials called with type {0}, itemgroup {1} and authentication {2}", new Object[] {type.getName(), itemGroup, authentication});
         if (ACL.SYSTEM.equals(authentication)) {
             ArrayList<C> list = new ArrayList<>();
-            for (IdCredentials credential : credentials.values()) {
+            for (KubernetesSourcedCredential credential : credentials.values()) {
+                // Parent group of item can be null
+                if(itemGroup == null && !credential.getItemGroups().isEmpty()) {
+                    continue;
+                }
+                if(itemGroup != null) {
+                    String itemGroupPath = '/' + itemGroup.getUrl();
+                    Collection<String> itemGroups = credential.getItemGroups();
+                    LOG.log(Level.FINEST, "getCredentials checking if itemGroupPath {0} is in itemGroups of {1} ({2})", new Object[] { itemGroupPath, credential.getId(), itemGroups });
+                    if (!itemGroups.isEmpty() && itemGroups.stream().noneMatch(itemGroupPath::startsWith)) {
```

Comment:
using string prefix matching to match a job hierachy when we can itterated the job hierachy and match exact strings makes me a little aprehensive.  esp when security is concerned.

whilst using a for loop here would increate the itteration time it may be safer in the long run.

That said I have a feeling this needs to change to support the UX where we show credentials in a store appropriate for the location(s) they are defined in?

Reply:
you are right, it makes sense to match the paths exactly as the stores pass on secrets to children anyways, it will then also show the correct store

Reply:
I think more work is needed to show it in the right store?
currently we have a [single store that is scoped to the root of Jenkins](https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/blob/95949f923aae4c2261bb5d187aeafb5e60683510/src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialsStore.java)

Reply:
I took care of that by adding context to the store and pass it on from the getStore function

Reply:
thanks I missed that.